pkg/client/drive_adapter_test.go

```diff
@@ -0,0 +1,693 @@
+	"github.com/minio/direct-csi/pkg/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	fakedynamic "k8s.io/client-go/dynamic/fake"
+)
+
+func newUnstructured(backendVersion, kind, name string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": fmt.Sprintf("%s/%s", directcsi.Group, backendVersion),
+			"kind":       kind,
+			"metadata": map[string]interface{}{
+				"namespace": metav1.NamespaceNone,
+				"name":      name,
+			},
+		},
+	}
+}
+
+func getFakeDirectCSIAdapter(backendVersion, resource string, object *unstructured.Unstructured) (*directCSIInterface, error) {
+	fakeResourceInterface := fakedynamic.NewSimpleDynamicClient(
+		runtime.NewScheme(),
+		object,
+	)
+	dynamicResourceClient := fakeResourceInterface.Resource(
+		schema.GroupVersionResource{
+			Group:    directcsi.Group,
+			Version:  backendVersion,
+			Resource: resource,
+		},
+	)
+	groupVersion := schema.GroupVersion{Group: directcsi.Group, Version: backendVersion}
+	return &directCSIInterface{resourceInterface: dynamicResourceClient, groupVersion: groupVersion}, nil
+}
+
+func getFakeDirectCSIDriveListAdapter(backendVersion, resource, value string, unstructuredObjects ...*unstructured.Unstructured) (*directCSIInterface, error) {
+	toRuntimeObjects := func(objs []*unstructured.Unstructured) (objects []runtime.Object) {
+		for _, unObj := range objs {
+			objects = append(objects, unObj)
+		}
+		return objects
+	}
+	fakeResourceInterface := fakedynamic.NewSimpleDynamicClientWithCustomListKinds(runtime.NewScheme(),
+		map[schema.GroupVersionResource]string{
+			{Group: directcsi.Group, Version: backendVersion, Resource: resource}: value,
+		},
+		toRuntimeObjects(unstructuredObjects)...,
+	)
+	dynamicResourceClient := fakeResourceInterface.Resource(
+		schema.GroupVersionResource{
+			Group:    directcsi.Group,
+			Version:  backendVersion,
+			Resource: resource,
+		},
+	)
+	groupVersion := schema.GroupVersion{Group: directcsi.Group, Version: backendVersion}
+	return &directCSIInterface{resourceInterface: dynamicResourceClient, groupVersion: groupVersion}, nil
+}
+
+func TestGetDrive(t *testing.T) {
+	testCases := []struct {
+		backendVersion string
+		kind           string
+		name           string
+		expectedDrive  *directcsi.DirectCSIDrive
+	}{
+		{
+			backendVersion: "v1alpha1",
+			kind:           "DirectCSIDrive",
+			name:           "drive-1",
+			expectedDrive: &directcsi.DirectCSIDrive{
+				TypeMeta: utils.DirectCSIDriveTypeMeta(),
+				ObjectMeta: metav1.ObjectMeta{Name: "drive-1", Labels: map[string]string{
+					"direct.csi.min.io/access-tier": "Unknown",
+					"direct.csi.min.io/created-by":  "directcsi-driver",
+					"direct.csi.min.io/node":        "",
+					"direct.csi.min.io/path":        "",
```

Comment:
We can test with non-empty label values, The labels will be populated based on this logic - https://github.com/minio/direct-csi/blob/master/pkg/converter/drive_upgrade.go#L149

These labels will be populated when the fields like `Status.NodeName`, `Status.Path` is set on the drives